### PR TITLE
chore: add stories for `MoreMenu`

### PR DESCRIPTION
**Pull request by aslilac**
Part of #11420
Part of #12261

### CODE REVIEW

File: site/src/components/MoreMenu/MoreMenu.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          </MoreMenuItem>
+        </MoreMenuContent>
+      </>
+    ),
+  },
+  play: async ({ canvasElement, step }) => {
+    const canvas = within(canvasElement);
+
+    await step("Open menu", async () => {
+      await userEvent.click(canvas.getByRole("button"));
```
Comment (Parkreiner) [inline site/src/components/MoreMenu/MoreMenu.stories.tsx:46]: To be clear, not blocking on this – it's more a note for myself for the accessibility backlog I'm slowly putting together:

It'd be nice if we could swap a selector like this in without things breaking:
```tsx
// Important thing is having any kind of accessible name - word choice doesn't matter
const menuButton = canvas.getByRole("button", {
  name: "Show options"
});
await userEvent.click(menuButton);
```

File: site/src/components/MoreMenu/MoreMenu.tsx [hunk 2]
```diff
@@ -111,11 +111,7 @@ export const MoreMenuItem: FC<MoreMenuItemProps> = ({
   danger = false,
   ...menuItemProps
 }) => {
-  const ctx = useContext(MoreMenuContext);
-
-  if (!ctx) {
-    throw new Error("MoreMenuItem must be used inside of MoreMenu");
-  }
+  const menu = useMoreMenuContext();
```
Comment (Parkreiner) [inline site/src/components/MoreMenu/MoreMenu.tsx:114]: Good catch on making sure this logic was centralized

**Review by buenos-nachos — COMMENTED** [review]
Looks good! Have one other question with the story output itself, but if you don't think it'll be a problem for now, I can approve

**Comment by Parkreiner** [comment]
There's something a little weird happening with the list items, where the indicator for the last-selected item with the arrow keys is conflicting with whichever list item is being hovered

https://github.com/coder/coder/assets/28937484/e08840ae-3b14-4d7e-81e1-4544ce4d90fa

I don't know what the plan is for the story rollout, so I'm happy to approve this regardless. My main concern is using Stories to cement functionality we know for a fact is wonky, but this might be the sort of thing we actually need to bring a Jest test in for

File: site/src/components/MoreMenu/MoreMenu.stories.tsx (hunk 1 above)
Comment (aslilac) [reply site/src/components/MoreMenu/MoreMenu.stories.tsx:46]: I didn't even know that was a thing you could do! the testing library API is kind of just weird and mystic to me 😅

**Comment by aslilac** [comment]
I've opened an issue to track the weird selection state so that we can take care of it later, but for now, since it's already in the app and no one has complained yet, I don't want to block adding the story on fixing it.

**Review by buenos-nachos — APPROVED** [review]
@aslilac Okay, cool. Approving

**Pull request merged**